pkg/options/types.go

```diff
@@ -104,6 +104,41 @@ func (r *ResourceSet) Type() string {
 	return "string"
 }
 
+// NodeNameType represents a nodeName to query from.
+type NodeNameType string
+
+// GetNodeNameFieldSelector returns a nodename field selector.
+func (n *NodeNameType) GetNodeNameFieldSelector() string {
+	if string(*n) != "" {
+		return fields.OneTermEqualSelector("spec.nodeName", string(*n)).String()
+	}
+	return fields.Nothing().String()
+}
+
+// EmptyFieldSelector returns an empty field selector.
+func EmptyFieldSelector() string {
+	return fields.Nothing().String()
+}
+
+// MergeFieldSelector returns AND of two field selectors.
+func MergeFieldSelector(s1 string, s2 string) (string, error) {
+	selector1, err := fields.ParseSelector(s1)
+	if err != nil {
+		return EmptyFieldSelector(), err
+	}
+	selector2, err := fields.ParseSelector(s2)
+	if err != nil {
+		return EmptyFieldSelector(), err
+	}
+	if selector1.Empty() {
+		return selector2.String(), nil
+	}
+	if selector2.Empty() {
+		return selector1.String(), nil
+	}
+	return fields.AndSelectors(selector1, selector2).String(), nil
```

Comment:
Is it safe to assume that we'd always be working with two selectors here? What if we add another filter in the future? Since this is not a method on `*NodeNameType` but a general function inside `options`, do we want to generalize this a bit further, so as to be able to merge `n` selectors?